Testing/functional/tests/cypress/support/functions/intercept.js

```diff
@@ -99,3 +111,291 @@ export function setupUserProfileIntercept(options = {}) {
+    ).as("getPatientDataForBcCancerScreening");
+    cy.intercept(
+        "GET",
+        "**/PatientData/*patientDataTypes=DiagnosticImaging*"
+    ).as("getPatientDataForDiagnosticImaging");
+    cy.intercept(
+        "GET",
+        "**/PatientData/*?patientDataTypes=OrganDonorRegistrationStatus*"
+    ).as("getOrganDonorRegistrationStatus");
+    cy.intercept("GET", "**/UserProfile/*").as("getUserProfile");
+}
+
+export function waitForInitialDataLoad(username, config, path) {
+    const featureToggle = config.webClient.featureToggleConfiguration;
+
+    cy.log(`Username: ${username}`);
+    cy.log(`Feature Toggle: ${JSON.stringify(featureToggle)}`);
+    cy.log(`Path: ${path}`);
+
+    cy.log("Wait on patient.");
+    cy.wait("@getPatient", { timeout: defaultTimeout });
+
+    waitForUserProfile(username).then((blockedDataSources) => {
+        waitForClinicalDocument(featureToggle, path, blockedDataSources);
+        waitForOrganDonorRegistratonStatusService(
+            featureToggle,
+            path,
+            blockedDataSources
+        );
+        waitForPatientDataForBcCancerScreening(
+            featureToggle,
+            path,
+            blockedDataSources
+        );
+        waitForPatientDataForDiagnosticImaging(
+            featureToggle,
+            path,
+            blockedDataSources
+        );
+    });
+
+    cy.log("Wait on communication.");
+    cy.wait("@getCommunication", { timeout: defaultTimeout });
+
+    waitForNotification(featureToggle);
+}
+
+function waitForUserProfile(username) {
+    return new Cypress.Promise((resolve) => {
+        let blockedDataSources;
+
+        if (
+            username !== Cypress.env("keycloak.deceased.username") &&
+            username !== Cypress.env("keycloak.accountclosure.username")
+        ) {
+            cy.log("Wait on user profile.");
+            cy.wait("@getUserProfile", { timeout: defaultTimeout }).then(
+                (interception) => {
+                    const responseBody = interception.response.body;
+                    blockedDataSources =
+                        responseBody.resourcePayload?.blockedDataSources;
+
+                    cy.log(
+                        `Get User Profile Blocked Data Sources: ${
+                            blockedDataSources
+                                ? JSON.stringify(blockedDataSources)
+                                : "Blocked data sources are not available"
+                        }`
+                    );
+
+                    resolve(blockedDataSources);
+                }
+            );
+        } else {
+            resolve(blockedDataSources);
+        }
+    });
+}
+
+function waitForNotification(featureToggle) {
+    cy.log(
+        `waitForNotification called - enabled: ${featureToggle.notificationCentre.enabled}`
+    );
+
+    if (
+        featureToggle.notificationCentre &&
+        featureToggle.notificationCentre.enabled
+    ) {
```

Comment:
Can be simplified: `featureToggle.notificationCentre?.enabled`

Reply:
Changed.